app/src/main/java/io/homeassistant/companion/android/widgets/ButtonWidget.kt

```diff
@@ -0,0 +1,135 @@
+package io.homeassistant.companion.android.widgets
+
+import android.app.PendingIntent
+import android.appwidget.AppWidgetManager
+import android.appwidget.AppWidgetProvider
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import android.widget.RemoteViews
+import io.homeassistant.companion.android.R
+import io.homeassistant.companion.android.common.dagger.GraphComponentAccessor
+import io.homeassistant.companion.android.domain.integration.IntegrationUseCase
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+
+class ButtonWidget : AppWidgetProvider() {
+    companion object {
+        private const val TAG = "ButtonWidget"
+        private const val CALL_SERVICE = "CALL_SERVICE"
+    }
+
+    private lateinit var domain: String
+    private lateinit var service: String
+    private lateinit var serviceData: String
+
+    @Inject
+    lateinit var integrationUseCase: IntegrationUseCase
+
+    private val mainScope: CoroutineScope = CoroutineScope(Dispatchers.Main + Job())
+
+    override fun onUpdate(
+        context: Context,
+        appWidgetManager: AppWidgetManager,
+        appWidgetIds: IntArray
+    ) {
+        // There may be multiple widgets active, so update all of them
+        for (appWidgetId in appWidgetIds) {
+            loadServiceCallData(context, appWidgetId)
```

Comment:
You load this but never use any of it in this method?

Reply:
Right, this was originally a loading method I had implemented before finding the update ordering bug noted in the ButtonWidgetConfigureActivity.kt lines 40-42.  I have cleaned this up and reworked loading in the latest commit.